### PR TITLE
Adding sample for non interactive oauth2 client credentials flow component

### PR DESCRIPTION
**Pull request by h2floh**
# Description

This is the corresponding sample application for the oauth2 client credential middleware component. It will guide the user through the creation of an Azure Active Directory application with application permissions on Microsoft Graph.

The demo app itself will call Microsoft Graph with the injected token from the middleware.

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] The sample code compiles correctly
* [X] You've tested new builds of the sample if you changed sample code
* [X] You've updated the sample's README if necessary

### CODE REVIEW

**Comment by h2floh** [comment]
@orizohar moved the sample PR here and renamed quickstart -> sample. 

CC @yaron2

File: middleware-clientcredentials/README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
+# OAuth Authorization to external service (non interactive)
+
+## Sample info
+| Attribute | Details |
+|--------|--------|
+| Dapr runtime version | v0.9 |
```
Comment (h2floh) [inline middleware-clientcredentials/README.md:6]: Is that be the correct runtime version?

Comment (orizohar) [reply middleware-clientcredentials/README.md:6]: This is the current version of date, if you created this sample with that version then yes. (you can double check with `dapr --version` on the machine you validated this with)

**Comment by orizohar** [comment]
@pruthvidhodda @yaron2 @amanbha - can you please do a code review?

**Comment by orizohar** [comment]
Also, can you please add as part of this PR an entry in the TOC of the main samples README? Here: https://github.com/dapr/samples#samples-in-this-repository

File: middleware-clientcredentials/README.md (hunk 1 above)
Comment (h2floh) [reply middleware-clientcredentials/README.md:6]: Ok, than that is the correct version.

File: middleware-clientcredentials/README.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- [Helm](https://github.com/helm/helm)
+- A working [Azure Active Directory] with Administrator rights, alternatively you can [create one](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-access-create-new-tenant)
+
+
+### Step 1 - Clone the sample repository
+
+1. Clone the sample repo, then navigate to the middleware sample:
+```bash
+git clone [-b <dapr_version_tag>] https://github.com/dapr/samples.git
```
[the remark this answers is not in the document]
Comment (h2floh) [reply middleware-clientcredentials/README.md]: Ok, removed the [-b ] attribute

File: middleware-clientcredentials/README.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+cd samples/middleware-clientcredentials/msgraphapp
+```
+> **Note**: See https://github.com/dapr/samples#supported-dapr-runtime-version for supported tags. Use `git clone https://github.com/dapr/samples.git` when using the edge version of dapr runtime.
```
[the remark this answers is not in the document]
Comment (h2floh) [reply middleware-clientcredentials/README.md]: Removed

File: middleware-clientcredentials/README.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+2. Examine the ```app.js``` file. You'll see this is a simple Node.js Express web server with a single ```/users``` route that returns the Microsoft Graph API result based on the input query parameter ```displayName```. Also you can see that we will forward the token saved in the request header called `msgraph-token` as the `Authorization` header in the request towards the MS Graph API.
```
[the remark this answers is not in the document]
Comment (h2floh) [reply middleware-clientcredentials/README.md]: Thanks! Changed the sentence and checked other occurrences!

**Comment by orizohar** [comment]
@h2floh - Thank you for your contribution!

**Pull request merged**